src/portfft/common/global.hpp

```diff
@@ -396,14 +418,140 @@ std::vector<sycl::event> compute_level(
                 it};
             dispatch_level<Scalar, LayoutIn, LayoutOut, SubgroupSize>(
                 &in_acc_or_usm[0] + input_batch_offset, offset_output, &in_imag_acc_or_usm[0] + input_batch_offset,
-                offset_output_imag, subimpl_twiddles, multipliers_between_factors, &loc_for_input[0],
+                offset_output_imag, subimpl_twiddles, load_modifier_data, store_modifier_data, &loc_for_input[0],
                 &loc_for_twiddles[0], &loc_for_modifier[0], factors_triple, inner_batches, inclusive_scan, batch_size,
                 global_data, kh);
           });
     }));
   }
   return events;
 }
+
+/**
+ * Run the global implementation
```

Comment:
What exactly is this function doing? It seems to be a lot of code added, while we could "Run the global implementation" before without it. 

Is this called from host or kernel?

Reply:
This is not new piece of code. This is exactly [this loop](https://github.com/codeplaysoftware/portFFT/blob/d5fe215de93e38244278909d5f2da8c5853e566f/src/portfft/dispatcher/global_dispatcher.hpp#L324) here which is refactored out into a different function.
Its called from host

Reply:
But that loop is still there as well. Overall this has me confused as the new version looks like much more code and it is unclear what the responsibility of each function is. If nothing else the documentation of this function needs to be improved. It needs to be clear in what way this is different from run_kernel_struct::execute.

Reply:
Sorry I was not clear with that comment.

What I mean was all the logic inside that loop which is linked, is moved out to a new function, as I need that logic twice (one for forward, and one for backward pass of Bluestein). See how it is called over [here](https://github.com/codeplaysoftware/portFFT/blob/4976174f0dd06c3fe162b7b715257e81d6b3827e/src/portfft/dispatcher/global_dispatcher.hpp#L404) and [here](https://github.com/codeplaysoftware/portFFT/blob/4976174f0dd06c3fe162b7b715257e81d6b3827e/src/portfft/dispatcher/global_dispatcher.hpp#L427)